modules/sov-ibc-transfer/src/lib.rs

```diff
@@ -28,22 +28,19 @@ pub struct IbcTransfer<S: Spec> {
     #[module]
     bank: sov_bank::Bank<S>,
 
-    /// Keeps track of the address of each token we minted by token denom.
+    /// Maps the token name to its corresponding token ID for tokens minted
+    /// through IBC. This mapping is used during mint/burn processes to validate
+    /// if the token exists and gives out needed ID for the minting/burning
+    /// process.
     #[state]
-    minted_tokens: StateMap<String, TokenId>,
+    minted_token_name_to_id: StateMap<String, TokenId>,
 
-    /// Keeps track of the address of each token we escrowed as a function of
-    /// the token denom. We need this map because we have the token ID
-    /// information when escrowing the tokens (i.e. when someone calls a
-    /// `send_transfer()`), but not when unescrowing tokens (i.e in a
-    /// `recv_packet`), in which case the only information we have is the ICS 20
-    /// denom, and amount. Given that every token that is unescrowed has been
-    /// previously escrowed, our strategy to get the token ID associated
-    /// with a denom is
-    /// 1. when tokens are escrowed, save the mapping `denom -> token ID`
-    /// 2. when tokens are unescrowed, lookup the token ID by `denom`
+    /// Maps the token ID to its corresponding token name for tokens minted
+    /// through IBC. This mapping is used during escrows/un-escrows to verify if
+    /// the `TokenId` extracted from the `denom` is not an IBC-minted token,
+    /// indicating it is a native token for escrow/un-escrows purposes.
     #[state]
-    escrowed_tokens: StateMap<String, TokenId>,
+    minted_token_id_to_name: StateMap<TokenId, String>,
```

Comment:
just a note: we don't really need to maintain `token_name`. we can use something like `StateSet`.

Reply:
That's unfortunate. We don't have the `StateSet` type. We could go with the `StateMap<TokenId, ()>`, though I opted for storing the name instead, if in case needed in queries later. Thought it shouldn't be an expensive decision.